### ISSUE TITLE
Rewrite the registration code

### ISSUE THREAD

**Issue by arwassa** [dotkom/regme#139]
State in the corrent implementation seems very wonky and behaves in strange ways, should just be rewritten.

**Comment by oleast** [comment]
@Aborysa Was this fixed with the services PR?

**Comment by arwassa** [comment]
No, not really. Some of the issues were resolved, but many of the components are way too large.
